src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -51,75 +75,140 @@ public bool IsEnabled(LogLevel logLevel)
             string details = string.Empty;
-            if (exception != null)
+            if (logItem.Exception != null)
             {
-                if (string.IsNullOrEmpty(functionName) && exception is FunctionInvocationException fex)
+                // Populate details from the exception.
+                if (string.IsNullOrEmpty(functionName) && logItem.Exception is FunctionInvocationException fex)
                 {
                     functionName = string.IsNullOrEmpty(fex.MethodName) ? string.Empty : fex.MethodName.Replace("Host.Functions.", string.Empty);
                 }
 
-                (innerExceptionType, innerExceptionMessage, details) = exception.GetExceptionDetails();
+                (innerExceptionType, innerExceptionMessage, details) = logItem.Exception.GetExceptionDetails();
                 innerExceptionMessage = innerExceptionMessage ?? string.Empty;
             }
 
-            _eventGenerator.LogFunctionTraceEvent(logLevel, subscriptionId, appName, functionName, eventName, source, details, summary, innerExceptionType, innerExceptionMessage, functionInvocationId, hostInstanceId, activityId, runtimeSiteName, slotName);
+            _eventGenerator.LogFunctionTraceEvent(logItem.LogLevel, _subscriptionId, _appName, functionName, eventName, source, details, summary, innerExceptionType, innerExceptionMessage, functionInvocationId, _hostInstanceId, activityId, _runtimeSiteName, _slotName);
+        }
+
+        private bool IsUserLog<TState>(TState state)
+        {
+            // User logs are determined by either the category or the presence of the LogPropertyIsUserLogKey
+            // in the log state.
+            // This check is extra defensive; the 'Function.{FunctionName}.User' category should never occur here
+            // as the SystemLoggerProvider checks that before creating a Logger.
+
+            return _isUserFunction ||
+                (state is IEnumerable<KeyValuePair<string, object>> stateDict &&
+                Utility.GetStateBoolValue(stateDict, ScriptConstants.LogPropertyIsUserLogKey) == true);
+        }
+
+        private void InitializeApplicationInfo()
+        {
+            _subscriptionId = _environment.GetSubscriptionId() ?? string.Empty;
+            _appName = _environment.GetAzureWebsiteUniqueSlotName() ?? string.Empty;
+            _runtimeSiteName = _environment.GetRuntimeSiteName() ?? string.Empty;
+            _slotName = _environment.GetSlotName() ?? string.Empty;
+        }
+
+        private void TimerFlush(object state)
+        {
+            if (_logItemQueue.Count > 0)
+            {
+                List<LogItem> currentLogItems = null;
+                lock (_syncLock)
+                {
+                    currentLogItems = _logItemQueue;
+                    _logItemQueue = new List<LogItem>();
+                }
+
+                foreach (var logItem in currentLogItems)
+                {
+                    LogCore(logItem);
```

Comment:
Logs are still written in the same order they were written in, however the timestamps will now be different. If we want, we could add an EventTimeStamp to capture when the event was actually written. We do something similar for Linux Consumption logs, which are also logged in batches external to the runtime.